CMakeLists.txt

```
@@ -75,13 +64,27 @@ option(use_tpm_simulator "tpm simulator type of hsm used with the provisioning c
 option(use_edge_modules "Enable support for running modules against Azure IoT Edge" OFF)
 option(use_custom_heap "use externally defined heap functions instead of the malloc family" OFF)
 option(use_cppunittest "set use_cppunittest to ON to build CppUnitTest tests on Windows (default is OFF)" OFF)
+option(enable_compiler_warnings_as_errors "enable strict compiler warnings-as-errors" ON)
```

```suggestion
option(warnings_as_errors "enable strict compiler warnings-as-errors" ON)
```

makes it easier on the command line